warehouse/query-core/src/main/java/datawave/core/iterators/TermFrequencyIterator.java

```diff
@@ -191,6 +191,10 @@ protected void findTop() throws IOException {
             }
             
             tfKey.parse(k);
+            if (!tfKey.isValid()) {
+                log.error("Malformed term frequency key: " + k.toStringNoTime());
+                source.next();
```

Comment:
do we really want to skip it?

Reply:
If we want to skip an invalid key, don't we want an explicit `continue` after `source.next()` to restart the while loop at 180? It is not clear that to me whether nothing will hapen in lines 199-209 below, with the worst case possibly calling source.next() again in the else clause.

Reply:
An explicit continue is the right thing to do here, I'll add that in.